### PR TITLE
Added an export option in Archival Insitutions and Descriptions to enable RiC conversions

### PR DESCRIPTION
**Pull request by herminiogg**
Hello,

As a first step in the Linked Open Data deployment I added a small convenient option for exporting in RiC the institutions and collections descriptions. In a nutshell, it converts the current description to RiC on the fly using ShExML. For doing so, it uses the same mapping rules developed for the batch conversion but adapted for a single item export. The mapping rules extract the data from the API and the already present XML export. 

Some further details about the implementation:
- Right now, the code is in the front-end as it does not need to query any additional data from the API. Whether or not this could be included in the API, could be a future discussion.
- In order to avoid some dependency collisions I upgraded jena-iri and commons-text to the same version that ShExML, and its transitive dependencies, are using. It seems that everything is working OK with the new versions.
- The mapping rules are externalised in two files, so in case that some adjustments are needed in the future we can modify them without recompiling.

After my first tests it seems that the method works fine and fast enough. This could complement the future batch approach with more up-to-date data so all data can be accessed as LOD right after being imported in the portal. 

Another thing that we can think about for the future is to use the exports code to offer content negotiation. So machine agents can retrieve is preferred data format.

Best regards,
Herminio

### CODE REVIEW

**Comment by mikesname** [comment]
Looks great! I haven't had a chance to test yet but could you add some tests to `test/integration/portal/PortalSpec.scala` please? I suppose just after the `"export repository contents as EAD zip"`.

File: build.sbt [hunk 1]
```diff
@@ -39,10 +39,13 @@ val backendDependencies = Seq(
...
   // IRI helper...
-  "org.apache.jena" % "jena-iri" % "3.9.0",
+  "org.apache.jena" % "jena-iri" % "3.17.0",
 
   // Ontology
-  "eu.ehri-project" % "ehri-definitions" % backendVersion
+  "eu.ehri-project" % "ehri-definitions" % backendVersion,
+
+  // Conversion to RDF
+  "com.herminiogarcia" % "shexml_2.12" % "0.3.1"
```
Comment (mikesname) [inline build.sbt:48]: Is this lib compatible with Scala 2.13?

**Comment by mikesname** [comment]
Right, I've had a closer look and while the patch looks very nice itself I'm worried about pulling in so many transitive dependencies, both the stuff we don't need here (MariaDB, MSSQL drivers, etc) but also the stuff that collides, particularly Jackson and friends. I'd really like to minimise dependencies like this that regularly cause transitive collisions since it can be a nightmare to figure things out if, e.g. a Play security fix causes an incompatible class change error or something.

(One minor example of this in practice is that the lib bundles slf4j-nop, which collides at runtime with Play's bundled logback-classic implementation.)

I'm wondering whether it might be nice to set the ShExML converter up as its own web service (if it isn't already) and sit that behind a reverse proxy so the URLs would be similar. This is basically what we do for the OAI-PMH service currently, except that's part of the Neo4j backend rather than a separate service.

This has some advantages in addition to the dependencies issue, not least it's much easier to update the conversion without releasing new versions of the portal. OTOH it's another service to deploy. What do you think?

**Comment by herminiogg** [comment]
Yes, that was something that crossed my mind while I was doing the implementation. Going for a microservice architecture could be indeed a way to avoid dependencies collisions and build more independent services. As I was trying the approach I didn't go this way but I totally agree. 

In fact, I have already a web service running for the ShExML playground (http://shexml.herminiogarcia.com/) but it is running on a small server so it is not intended for heavy use. However, we can use the existing code (https://github.com/herminiogg/shexml-webapp) to deploy an instance on EHRI servers. If you want to have the mapping rules on the server, then we should create the required wrapping code in the webservice so the call is really simple for the EHRI portal.

File: build.sbt (hunk 1 above)
Comment (herminiogg) [reply build.sbt:48]: Not yet, I always wanted to enable the cross compilation but never really had the time to do it. If you plan to update I can try to enable it for 2.13. Nevertheless, for Scala 3 that's another story due to the dependencies that are still not updated.

**Comment by mikesname** [comment]
Sounds good. The approach I have in mind uses a reverse proxy so the only changes to the portal required would be to add links to the relevant pages. Then in the Apache config we would do something like this:

    ProxyPassMatch "/institutions/([^/]+)/export/ric"  "https://shexml.ehri-project.eu/convert?map=repository&src=https://portal.ehri-project.eu/api/v1/$1"

That assumes we set up a ShExML convert service with the required mapping data able to accept input JSON from a URL.

Ideally the web service would be a Play app because I've already figured out how to deploy them ;)

**Comment by herminiogg** [comment]
It sounds good! I think it should not take a lot of time to do the web service in Play. I will try to refresh my knowledge of Play and dedicate some time to it!

**Pull request closed without merging**